echo_server.py

```diff
@@ -50,22 +52,45 @@ def response_error(error_num, error_msg):
 
 def parse_request(request):
     request_list = request.split()
-    print len(request_list)
-    if len(request_list) != 5:
-        return None
     error_check = check_for_errors(request_list)
     if error_check == "Good to go!":
-        return response_ok(request_list[1])
+        body_and_type = resolve_uri(request_list[1])
+        return response_ok(request_list[1], body_and_type)
     return error_check
 
 
 def check_for_errors(request):
+    if len(request) < 5:
+        return response_error('400', 'BAD REQUEST')
     if request[0] != 'GET':
         return response_error('405', '{} METHOD NOT ALLOWED'.format(request[0]))
+    if not os.path.exists(request[1]):
+        return response_error('404', 'NOT FOUND')
```

Comment:
This seems an odd place to do this work.  Checking if the file path exists is much more a job associated with the work in `resolve_uri.  Putting it here means that you have a mix of concerns and that you are making assumptions in resolve_uri that may not be true when testing.